pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -5012,9 +5012,7 @@ protected CompletableFuture<Void> internalRemoveSubscribeRate(boolean isGlobal)
 
     protected void handleTopicPolicyException(String methodName, Throwable thr, AsyncResponse asyncResponse) {
         Throwable cause = thr.getCause();
-        if (!(cause instanceof WebApplicationException) || !(
-                ((WebApplicationException) cause).getResponse().getStatus() == 307
-                        || ((WebApplicationException) cause).getResponse().getStatus() == 404)) {
+        if (!isRedirectException(cause) && !isNotFoundException(cause)) {
```

Comment:
Adding a new method `isNotRedirectAndNotFoundException` in this class may be better.

Reply:
Fixed. Since the `404` error named `NotFound` may make this name confuse, I named a new method `isNot307And404Exception`.

Reply:
@lhotari Is the new method ok ?

Reply:
@codelipenghui is this new method name ok ?